src/gui/guiChatConsole.cpp

```diff
@@ -639,3 +679,109 @@ void GUIChatConsole::setVisible(bool visible)
 	}
 }
 
+// Return how many "ctrl" keycodes successfully found in string, or 0 on fail
+int GUIChatConsole::setupChatClickCtrlKeys(std::string inputline)
+{
+	m_cache_chat_weblink_ctrl_keys.clear();
+
+	irr::EKEY_CODE kc;
+	std::string stemp;
+	size_t startpos = 0, endpos = 0;
+	while(startpos < inputline.size())
+	{
+		// Foreach delimited string,
+		endpos = inputline.find(',', startpos);
+		endpos = std::min(endpos, inputline.find(' ', startpos));
+		// Ignore space/comma
+		if(endpos == startpos)
+		{
+			++endpos;
+		}
+		// Ignore consecutive space/comma
+		else if(endpos - startpos > 1)
+		{
+			// If valid keycode, add it to cached list
+			stemp = inputline.substr(startpos, endpos - startpos);
+			kc = keyname_to_keycode_safemode(stemp.c_str());
+			if(kc != irr::KEY_KEY_CODES_COUNT)
+			{
+				m_cache_chat_weblink_ctrl_keys.push_back(kc);
+			}
+			else
+			{
+				stemp = "Ignoring unknown keycode '" + stemp + "' for chat_weblink_ctrl_keys, check your conf";
+				g_logger.log(LL_WARNING, stemp);
```

Comment:
I don't think this is necessary. Players will notice when this does not work; and no other key read-in produces such log messages (hence inconsistent).